testing/upload_validation.ts

```diff
@@ -933,138 +942,122 @@ const genericObjectSchema: z.ZodTypeAny = z.lazy(() =>
     .superRefine((data, context) => {
       const schema = data as GenericObjectSchema;
 
-      const validateTitleProperty = () => {
-        if (schema.titleProperty) {
-          if (!(schema.titleProperty in schema.properties)) {
+      function validateProperty(
+        propertyKey: keyof typeof schema,
+        isValidSchema: (schema: Schema & ObjectSchemaProperty) => boolean,
+        invalidSchemaMessage: string,
+      ) {
+        if (schema[propertyKey]) {
+          const propertyValue =
+            typeof schema[propertyKey] === 'string'
+              ? (schema[propertyKey] as string)
```

Comment:
Hmm, surprised these casts are needed, would expect TS to properly narrow these types. It may work without the casts if you store `schema[propertyKey]` into a local var and check that instead.